lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java

```diff
@@ -943,4 +943,78 @@ public void detach_success_callback_detached() throws AblyException {
 				ably.close();
 		}
 	}
+
+	/**
+	 * <p>
+	 * Validate publish will result in an error, when the channel is
+	 * in or moves to the FAILED state before the operation succeeds
+	 * </p>
+	 * <p>
+	 * Spec: RTL6c3
+	 * </p>
+	 *
+	 * @throws AblyException
+	 */
+	@Test
```

Comment:
> It gets executed, if we call publish on already failed channel.

ok, my mistake.

> Would you like me to add test cases for that also?

Not in this PR, thanks :)

However ... lets look at the comments at the top of this test:

````
Validate publish will result in an error, when the channel is
in or moves to the FAILED state before the operation succeeds
````

The `... when the channel is in ... the FAILED state` part is not part of this test, so please update the comment to reflect that.

Also, this test does not verify that the `publish()` results in an error. If verifies that the channel implicit attach is triggered, and the channel reaches the `failed` state, but it does not look at the result of the `publish()`. Please add a `CompletionListener` to the `publish()` call, and verify that there was an error. Also please review the tests below for the same issue.